packages/devtools/lib/src/memory/memory.dart

```diff
@@ -285,9 +293,214 @@ class MemoryScreen extends Screen with SetStateMixin {
 
     _updateStatus(null);
 
+    vmMemorySnapshotButton.disabled = true;
+    memoryController.computeLibraries().then((_) {
+      // Enable snapshot/setting buttons, all libraries/classses have been seen.
+      vmMemorySnapshotButton.disabled = false;
+      settingsButton.disabled = false;
+    });
+
     return screenDiv;
   }
 
+  TextField classNameFilter;
+  List<CoreElement> privateClasses;
+
+  /// Create the settings dialog.
+  void createSettingsDialog() {
+    settings = div(c: 'section settings-box')
+      ..add(<CoreElement>[
+        h2(text: 'Settings', c: 'settings-title'),
+        div(c: 'settings-area')
+          ..add(<CoreElement>[
+            form()
+              ..layoutHorizontal()
+              ..clazz('align-items-center')
+              ..add(<CoreElement>[
+                div(
+                  text: 'Display Snapshot Classes from Library',
+                  c: 'collapsible-885 flex-no-wrap settings-left',
+                )..add([
+                    librariesUi = div(c: 'settings-libraries'),
+                  ]),
+                div(c: 'setttings-options')
+                  ..add([
+                    div(c: 'filter-pattern-area')
+                      ..add(<CoreElement>[
+                        span(
+                          text: 'Class Name Pattern: ',
+                          c: 'settings-class-pattern',
+                        ),
+                        classNameFilter = TextField(c: 'filter-class')
```

Comment:
you could set the default value to show if this is empty as "any"
to indicate all classes are allowed

Reply:
Good idea, done.